pkgs/development/python-modules/pandera/default.nix

```diff
@@ -0,0 +1,124 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  multimethod,
+  numpy,
+  packaging,
+  pandas,
+  pydantic,
+  typeguard,
+  typing-inspect,
+  wrapt,
+  # test
+  joblib,
+  pyarrow,
+  pytestCheckHook,
+  pytest-asyncio,
+  # optional dependencies
+  black,
+  dask,
+  fastapi,
+  geopandas,
+  hypothesis,
+  pandas-stubs,
+  polars,
+  pyyaml,
+  scipy,
+  shapely,
+}:
+
+buildPythonPackage rec {
+  pname = "pandera";
+  version = "0.20.4";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "unionai-oss";
+    repo = "pandera";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-VetLfZlBWok7Mr1jxlHHjDH/D5xEsPFWQtX/hrvobgQ=";
+  };
+
+  build-system = [ setuptools ];
+
+  dependencies = [
+    multimethod
+    numpy
+    packaging
+    pandas
+    pydantic
+    typeguard
+    typing-inspect
+    wrapt
+  ];
+
+  optional-dependencies =
+    let
+      dask-dataframe = [ dask ] ++ dask.optional-dependencies.dataframe;
+      extras = {
+        strategies = [ hypothesis ];
+        hypotheses = [ scipy ];
+        io = [
+          pyyaml
+          black
+          #frictionless
+        ];
+        #pyspark = [ pyspark ] ++ pyspark.optional-dependencies.connect;
+        #modin = [ # modin not in nixpkgs
+        #  modin
+        #  ray
+        #  dask-dataframe
+        #];
+        #modin-ray = [
+        #  modin
+        #  ray
+        #];
+        #modin-dask = [
+        #  modin
+        #  dask-dataframe
+        #];
```

Comment:
Are these packages optional? Why are they commented out?

Reply:
Yes, they're optional, hence in optional-dependencies; most are commented out because `modin` is not in Nixpkgs as per the comment; `pyspark` is commented out because it currently does not define `optional-dependencies.connect`.  I'll add a comment to that effect.

Reply:
Done